samples/test/createStartInstance.test.js

```diff
@@ -0,0 +1,215 @@
+  });
+
+  it('should create instance from snapshot', async () => {
+    const projectId = await instancesClient.getProjectId();
+
+    const [newestDebian] = await imagesClient.getFromFamily({
+      project: 'debian-cloud',
+      family: 'debian-10',
+    });
+
+    await createDisk(projectId, zone, diskName, newestDebian.selfLink);
+    await createDiskSnapshot(projectId, zone, diskName, snapshotName);
+
+    const diskSnapshotLink = `projects/${projectId}/global/snapshots/${snapshotName}`;
+
+    const output = execSync(
+      `node instances/create-start-instance/createInstanceFromSnapshot ${projectId} ${zone} ${instanceName} ${diskSnapshotLink}`
+    );
+    assert.match(output, /Instance created./);
+
+    execSync(`node deleteInstance ${projectId} ${zone} ${instanceName}`);
+
+    await deleteDiskSnapshot(projectId, snapshotName);
+    await deleteDisk(projectId, zone, diskName);
+  });
+
+  it('should create instance with snapshotted data disk', async () => {
+    const projectId = await instancesClient.getProjectId();
+
+    const [newestDebian] = await imagesClient.getFromFamily({
+      project: 'debian-cloud',
+      family: 'debian-10',
+    });
+
+    await createDisk(projectId, zone, diskName, newestDebian.selfLink);
+    await createDiskSnapshot(projectId, zone, diskName, snapshotName);
+
+    const diskSnapshotLink = `projects/${projectId}/global/snapshots/${snapshotName}`;
+
+    const output = execSync(
+      `node instances/create-start-instance/createInstanceWithSnapshottedDataDisk ${projectId} ${zone} ${instanceName} ${diskSnapshotLink}`
+    );
+    assert.match(output, /Instance created./);
+
+    execSync(`node deleteInstance ${projectId} ${zone} ${instanceName}`);
+
+    await deleteDiskSnapshot(projectId, snapshotName);
+    await deleteDisk(projectId, zone, diskName);
+  });
+
+  it('should create instance from custom image', async () => {
+    const projectId = await instancesClient.getProjectId();
+
+    const [newestDebian] = await imagesClient.getFromFamily({
+      project: 'debian-cloud',
+      family: 'debian-10',
+    });
+
+    const output = execSync(
+      `node instances/create-start-instance/createInstanceFromCustomImage ${projectId} ${zone} ${instanceName} ${newestDebian.selfLink}`
+    );
+    assert.match(output, /Instance created./);
+
+    execSync(`node deleteInstance ${projectId} ${zone} ${instanceName}`);
+  });
+
+  it('should create instance with additional disk', async () => {
+    const projectId = await instancesClient.getProjectId();
+
+    const output = execSync(
+      `node instances/create-start-instance/createInstanceWithAdditionalDisk ${projectId} ${zone} ${instanceName}`
+    );
+    assert.match(output, /Instance created./);
+
+    execSync(`node deleteInstance ${projectId} ${zone} ${instanceName}`);
+  });
+
+  it('should create instance with subnet', async () => {
+    const projectId = await instancesClient.getProjectId();
```

Comment:
The problem with performing the delete step as part of a test, is that integration tests like this tend to be somewhat flaky (and in these edge cases we'll leak compute instances).

A pattern we've used for other long lived resources is to use a well known prefix for the instance we're creating, e.g., `test-instance-abc123`, and to then periodically delete any instances older than a given number of hours. Here's an example:

https://github.com/googleapis/nodejs-bigtable/blob/80022f846081a67c610870cec0e26f1da76d0dd5/samples/test/util.js#L36